File: CSharp/Product.cs
using System.Globalization;

namespace DevSuperior {

    class Product {

        public string Name {get; set;}
        public double Price {get; set;}
        public int Quantity {get; set;}
        public Product(string name, double price, int quantity) {

            this.Name = name;
            this.Price = price;
            this.Quantity = quantity;
        }

        public double total() {

            return Price * Quantity;
        }

        public void updatePrice(double percentage) {

            Price = Price * (1 + percentage / 100.00);
        }

        public override string ToString() {

            return Name
            + ", R$ "
            + Price.ToString("F2", CultureInfo.InvariantCulture)
            + ", "
            + Quantity;
        }
    }
}

File: Topico inicial/Java/br/com/app/Product.java
package br.com.app;

@Getter
@Setter
public class Product {

    private String name;
    private double price;
    private int quantity;

    public Product(String name, double price, int quantity) {

        this.name = name;
        this.price = price;
        this.quantity = quantity;
    }

    public double total(Product product) {

        return price * quantity;
    }

    public void updatePrice(double percentage) {

        price = price * (1 + percentage / 100.0);
    }

    @Override
    public String toString() {

        return name
        + ", $"
        + String.format("%.f2", price)
        + ", "
        + quantity;        
    }
}